### PR TITLE
[msan] Unwind stack before fatal reports

### PR DESCRIPTION
**Pull request by vitalybuka**
Msan does not unwind stack in malloc without origins, but we still need trace
for fatal errors.

### CODE REVIEW

File: compiler-rt/lib/msan/msan.h [hunk 1]
```diff
@@ -321,6 +321,9 @@ const int STACK_TRACE_TAG_VPTR = STACK_TRACE_TAG_FIELDS + 1;
     stack.Unwind(pc, bp, nullptr, common_flags()->fast_unwind_on_fatal); \
   }
 
+#define GET_FATAL_STACK_TRACE \
```
Comment (MaskRay) [inline compiler-rt/lib/msan/msan.h:324]: Add a comment that this is to complement `GET_MALLOC_STACK_TRACE` when not tracking origins.

**Comment by MaskRay** [comment]
In some functions there is a parameter `*stack` and a variable `stack` in a scope. I wonder whether the variable name for `FATAL` should be changed to reduce confusion...

**Review by MaskRay — APPROVED** [review]

File: compiler-rt/lib/msan/msan.h (hunk 1 above)
Comment (vitalybuka) [reply compiler-rt/lib/msan/msan.h:324]: I think it's redundant here, we have similar macro in other sanitizers, and it's just shortcut for GET_FATAL_STACK_TRACE_PC_BP.

Maybe these way as in new snapshot?

**Comment by vitalybuka** [comment]
> In some functions there is a parameter `*stack` and a variable `stack` in `{}` that shadows the parameter. I wonder whether the variable name for `FATAL` should be changed to reduce confusion...

That's kind of convention for all sanitizers, if we have macro, it create `stack` local object.
I am not rely concerned with shadowing, at it's only for fatal cases, but if you strong about that, I'd rather change parameters in allocator.

**Comment by MaskRay** [comment]
> I am not rely concerned with shadowing, at it's only for fatal cases, but if you strong about that, I'd rather change parameters in allocator.

Ah, I'm not concerned. You make the call whether the parameter or the local variable (both named `stack`) should be renamed to avoid confusion.

**Review by MaskRay — APPROVED** [review]

**Pull request merged**